pkg/catalog/routes.go

```diff
@@ -514,9 +529,150 @@ func (mc *MeshCatalog) listMeshServices() []service.MeshService {
 	return services
 }
 
+func (mc *MeshCatalog) destinationServicesFromTrafficTarget(t *target.TrafficTarget) ([]service.MeshService, error) {
+	sa := service.K8sServiceAccount{
+		Name:      t.Spec.Destination.Name,
+		Namespace: t.Spec.Destination.Namespace,
+	}
+	destServices, err := mc.GetServicesForServiceAccount(sa)
+	if err != nil {
+		return nil, errors.Errorf("Err finding Services for Service Account %#v: %v", sa, err)
+	}
+	return destServices, nil
+}
+
+func (mc *MeshCatalog) getHostnamesBasedOnNamespace(meshService service.MeshService) ([]string, []string, error) {
+	svc := mc.kubeController.GetService(meshService)
+	if svc == nil {
+		return nil, nil, errors.Errorf("Error fetching service %q", meshService)
+	}
+
+	hostnamesSameNamespace := kubernetes.GetLocalNamespaceHostnamesForService(svc)
+	hostnamesDiffNamespace := kubernetes.GetNamespaceScopedHostnamesForService(svc)
+
+	return hostnamesSameNamespace, hostnamesDiffNamespace, nil
+}
+
+func (mc *MeshCatalog) buildInboundPolicies(t *target.TrafficTarget) []*trafficpolicy.InboundTrafficPolicy {
+	inboundPolicies := []*trafficpolicy.InboundTrafficPolicy{}
+
+	// fetch services running workloads with destination service account
+	destServices, err := mc.destinationServicesFromTrafficTarget(t)
+	if err != nil {
+		log.Error().Msgf("Error resolving destination from traffic target %s (%s): %v", t.Name, t.Namespace, err)
+		return inboundPolicies
+	}
+
+	// fetch all routes referenced in traffic target
+	routeMatches, err := mc.routesFromRules(t.Spec.Rules, t.Namespace)
+	if err != nil {
+		log.Error().Msgf("Err finding route matches from TrafficTarget %s in namespace %s: %v", t.Name, t.Namespace, err)
+		return inboundPolicies
+	}
+
+	for _, destService := range destServices {
+		wc := getDefaultWeightedClusterForService(destService)
+
+		hostnamesInNS, hostnamesNotInNS, err := mc.getHostnamesBasedOnNamespace(destService)
+		if err != nil {
+			continue
+		}
+
+		// create traffic policy with domains without namespace suffix
+		policyInNS := trafficpolicy.NewInboundTrafficPolicy(policyName(destService, true), hostnamesInNS)
+		// create traffic policy for domains with namespace suffix
+		policyNotInNS := trafficpolicy.NewInboundTrafficPolicy(policyName(destService, false), hostnamesNotInNS)
```

Comment:
Why are 2 separate policies being created? I understand these are being mapped to virtual hosts but I am unsure why there needs to be 2 of these here.

Reply:
+1

Reply:
Take the sample scenario below. If you have two clients (one in the same ns and one outside of the ns) trying to access a destination on different routes (traffic targets configured), you'll need to differentiate the inbound policies for the hostnames local to the ns and the namespaced scoped hostnames like so:
```
Bookbuyer in bookstore namespace
Bookbuyer in bookbuyer namespace
Bookstore in bookstore namespace 


TrafficTarget:
bookbuyer.bookbuyer → Bookstore.bookstore on the /buy route
Bookbuyer.bookstore → bookstore.bookstore on the /sell route


VirtualHosts on bookstore:
{
Domains: bookstore, bookstore:8888,
Rules:
	bookbuyer.bookstore SA can access /buy route
}
{
Domains: bookstore.bookstore, bookstore.bookstore:8888, bookstore.bookstore.svc, etc.
Rules:
	Bookbuyer.bookbuyer SA can access /sell route
	Bookbuyer.bookstore SA can access /buy route
}

````

Reply:
@michelleN thanks for sharing the example. RBAC for the routes will be associated with routes, not the virtual host. So I would imagine the above example to look as follows:
```
Bookbuyer in bookstore namespace
Bookbuyer in bookbuyer namespace
Bookstore in bookstore namespace 


TrafficTarget:
bookbuyer.bookbuyer → Bookstore.bookstore on the /buy route
bookbuyer.bookstore → bookstore.bookstore on the /sell route


Inbound policies for bookstore.bookstore:
{
domains: bookstore, bookstore:8888, bookstore.bookstore, bookstore.bookstore:8888, bookstore.bookstore.svc, etc.
routes:
  - /buy
     rbac: source=bookbuyer.bookbuyer

  - /sell
     rbac: source=bookbuyer.bookstore
}
```

Essentially, the domains in the virtual_host don't need to be unique per route.

Reply:
In this example you would be allowing bookbuyer.bookbuyer SA to make requests to the buy path even if the host header on the request is "bookstore" which it cannot be because it is in a different namespace so it would never get routed here and that makes sense to me.

Reply:
@snehachhabria and I had a discussion around these lines as well earlier and she also recommended building one inbound policy per service. I'll go ahead and update the code as well. If we end up running into scenarios later (wondering if we might need this logic for ingress but not sure), we can add this logic back in if needed.